protocol/v2/ssv/validator/msgqueue_consumer.go

```diff
@@ -6,25 +6,32 @@ import (
 
 	specqbft "github.com/bloxapp/ssv-spec/qbft"
 	spectypes "github.com/bloxapp/ssv-spec/types"
-	"github.com/patrickmn/go-cache"
 	"go.uber.org/zap"
 
 	"github.com/bloxapp/ssv/protocol/v2/message"
-	"github.com/bloxapp/ssv/protocol/v2/ssv/msgqueue"
+	"github.com/bloxapp/ssv/protocol/v2/ssv/queue"
 )
 
 // MessageHandler process the msg. return error if exist
 type MessageHandler func(msg *spectypes.SSVMessage) error
 
 // HandleMessage handles a spectypes.SSVMessage.
 func (v *Validator) HandleMessage(msg *spectypes.SSVMessage) {
-	fields := []zap.Field{
+	v.logger.Debug("got message, pushing to queue",
 		zap.Int("queue_len", v.Q.Len()),
 		zap.String("msgType", message.MsgTypeToString(msg.MsgType)),
 		zap.String("msgID", msg.MsgID.String()),
+	)
+	decodedMsg, err := queue.DecodeSSVMessage(msg)
```

Comment:
heavy process? if so, maybe worth to do it async in order not to block network

Reply:
Yeah it's heavy, but decoding is expected to move to P2P layer (for validation).

In the short term, if this will cause slowdown, we can move decoding to a separate goroutine (as temporary solution until decoding moves to P2P layer).

Reply:
Yeah it's heavy, but decoding is expected to move to P2P layer (for validation).

In the short term, if this will cause slowdown, we can move decoding to a separate goroutine (as temporary solution until decoding moves to P2P layer).

Reply:
@lior-blox @amir-blox FYI